akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -514,46 +617,43 @@ private[akka] class Shard(
   private def receiveTerminated(ref: ActorRef): Unit = {
     if (handOffStopper.contains(ref))
       context.stop(self)
-    else if (idByRef.contains(ref) && handOffStopper.isEmpty)
-      entityTerminated(ref)
   }
 
   @InternalStableApi
-  def entityTerminated(ref: ActorRef): Unit = {
+  def entityTerminated(ref: ActorRef, id: EntityId): Unit = {
```

Comment:
We need to remember to talk to telemetry team about this

Reply:
We have the shard ctr as well

Reply:
Added a note to the board